staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -194,11 +195,28 @@ func (d *Helper) DeleteOrEvictPods(pods []corev1.Pod) error {
 
 func (d *Helper) evictPods(pods []corev1.Pod, policyGroupVersion string, getPodFn func(namespace, name string) (*corev1.Pod, error)) error {
```

Comment:
Would it make sense to also allow a user-provided context as well? In case there is a reason to stop progress for something other than hitting the globalTimeout?

Reply:
Yeah, I suggested the same in the PR's first comment.  But that would be a larger refactor and out of scope for this PR.

Reply:
I'm wondering if context should be plumbed through more of these methods, but not directly related to this PR.